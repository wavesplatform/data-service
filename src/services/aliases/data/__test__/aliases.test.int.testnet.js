const { createPgDriver } = require('../../../../db');
const loadConfig = require('../../../../loadConfig');
const options = loadConfig();
const create = require('../../index');

const ADDRESS = '3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek';

describe('Aliases', () => {
  const service = create({
    drivers: { pg: createPgDriver(options) },
    emitEvent: () => () => null,
  });

  describe('request by alias', () => {
    it('should return Alias for one correctly', done => {
      service
        .get('test-alias')
        .run()
        .listen({
          onResolved: alias => {
            expect(alias).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return null for non existing alias', done => {
      service
        .get('NON_EXISTING_ALIAS')
        .run()
        .listen({
          onResolved: nullable => {
            expect(nullable).toEqual(null);
            done();
          },
        });
    });
  });

  describe('request by address', () => {
    it('should return List(Alias) if requested without showBroken', done => {
      service
        .search({ address: ADDRESS })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });

    it('should return List(Alias) if requested with showBroken', done => {
      service
        .search({
          address: ADDRESS,
          showBroken: true,
        })
        .run()
        .listen({
          onResolved: mxs => {
            expect(mxs).toMatchSnapshot();
            done();
          },
        });
    });
  });
});
