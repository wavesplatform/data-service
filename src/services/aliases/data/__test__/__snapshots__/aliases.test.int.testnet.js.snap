// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aliases request by address should return List(Alias) if requested with showBroken 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-2",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-3",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-4",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-5",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-6",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-7",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test@test.test",
      },
    },
  ],
}
`;

exports[`Aliases request by address should return List(Alias) if requested without showBroken 1`] = `
Object {
  "__type": "list",
  "data": Array [
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-2",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-3",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-4",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-5",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-6",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test-alias-7",
      },
    },
    Object {
      "__type": "alias",
      "data": Object {
        "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
        "alias": "test@test.test",
      },
    },
  ],
}
`;

exports[`Aliases request by alias should return Alias for one correctly 1`] = `
Object {
  "__type": "alias",
  "data": Object {
    "address": "3NAkqT9JJHLF7hZyiZ5SncWsW8SQh2duZek",
    "alias": "test-alias",
  },
}
`;
